stl/inc/format

```diff
@@ -1006,18 +1008,19 @@ template <class _Context, class _Ty>
 
 template <class _Context, class _Ty>
 inline constexpr size_t _Get_format_arg_storage_size = sizeof(
-    _Get_format_arg_storage_type<_Context>(_STD declval<_Ty>()));
+    _Get_format_arg_storage_type<_Context>(static_cast<const _Ty&>(_STD declval<_Ty>())));
 
 struct _Format_arg_store_packed_index {
     // TRANSITION, Should be templated on number of arguments for even less storage
     using _Index_type = size_t;
 
     constexpr _Format_arg_store_packed_index() = default;
     constexpr explicit _Format_arg_store_packed_index(const size_t _Index)
-        : _Index(static_cast<_Index_type>(_Index)), _Type(_Basic_format_arg_type::_None) {}
+        : _Index(static_cast<_Index_type>(_Index)), _Type(static_cast<_Index_type>(_Basic_format_arg_type::_None)) {}
+
 
     _Index_type _Index : (sizeof(_Index_type) * 8 - 4);
-    _Basic_format_arg_type _Type : 4;
+    _Index_type _Type : 4;

```

Comment:
What was wrong here? I thought it should have packed those

Reply:
I think the general rule MSVC uses for bitfields is unrelated types don't pack well together, and it seems only equal size primitives pack together.